.circleci/config.yml

```diff
@@ -0,0 +1,121 @@
+version: 2.1
+
+jobs:
+  rspec:
+    parameters:
+      ruby-version:
+        type: string
+      gemfile:
+        type: string
+      database:
+        type: string
+    docker:
+      - image: circleci/ruby:<< parameters.ruby-version >>-node-browsers
+        environment:
+          DB: << parameters.database >>
+      - image: circleci/mysql:5.7-ram
+        environment:
+          MYSQL_DATABASE: avatax
+          MYSQL_USER: avatax
+          MYSQL_PASSWORD: password
+      - image: circleci/postgres:9.6-alpine
+        environment:
+          POSTGRES_USER: 'avatax'
+          POSTGRES_PASSWORD: 'password'
+          POSTGRES_DB: 'avatax'
+    steps:
+      - checkout
+      - when:
+          condition:
+            equal: ["mysql", << parameters.database >>]
+          steps:
+            - run:
+                name: Configure MySQL2
+                # Hack to set environment variable for later steps https://circleci.com/docs/2.0/env-vars/#example-configuration-of-environment-variables
+                command: echo 'export DATABASE_URL="mysql2://avatax:password@127.0.0.1/avatax"' >> $BASH_ENV
+      - when:
+          condition:
+            equal: ["postgres", << parameters.database >>]
+          steps:
+            - run:
+                name: Configure Postgres
+                command: echo 'export DATABASE_URL="postgresql://avatax:password@127.0.0.1/avatax"' >> $BASH_ENV
+      - run:
+          name: Configure bundler
+          command: bundle config --local path '~/.bundle'
+      - run:
+          name: Select Gemfile
+          command: cp gemfiles/<< parameters.gemfile >>.gemfile gemfiles/ci.gemfile
+      - restore_cache:
+          keys:
+            - bundler-v7-<< parameters.ruby-version >>-<< parameters.gemfile >>-{{ checksum "gemfiles/ci.gemfile" }}
+            - bundler-v7-<< parameters.ruby-version >>-<< parameters.gemfile >>
+      - run:
+          name: Install dependencies
+          command: bundle install --path=~/.bundle --gemfile=gemfiles/ci.gemfile
+      - save_cache:
+          key: bundler-v7-<< parameters.ruby-version >>-<< parameters.gemfile >>-{{ checksum "gemfiles/ci.gemfile" }}
+          paths:
+            - "~/.bundle"
+      - run:
+          name: Setup test app
+          command: bundle exec rake test_app
+      - run:
+          name: Run RSpec
+          command: bundle exec rake spec || [ "<< parameters.gemfile >>" = "spree_main" ]
+
+workflows:
+  all-specs:
+    jobs:
+      - rspec:
+          matrix:
+            parameters:
+              database:
+                - mysql
+                - postgres
+              gemfile:
+                - spree_3_1
+                - spree_3_7
+                - spree_4_0
+                - spree_4_1
+                - spree_4_2
+                - spree_main
```

Comment:
@agnieszkajacek let's please drop 3.1, 3.7,4.0 &  4.1. Only 4.2 & 4.3

Reply:
@damianlegawiec it looks like spree_avatax_official does not work with spree 4.3 :thinking: 
tests fails with: 
```
ActiveRecord::RecordInvalid:
        Validation failed: Stores is invalid
```

it occures with these combinations in both databases:
spree_4_3-3.0
spree_4_3-2.7

can it fail or not?

Reply:
@damianlegawiec :point_up: